grantnav/frontend/views.py

```diff
@@ -622,6 +632,50 @@ def widget_search(request, json_query):
     return JsonResponse(json_response)
 
 
+def reorder_recipient_org_age_when_awarded(context):
+    """ Put the ES results into a fixed order determined by AGE_BIN_LABELS
+        Note this is not a sort - it's a specific ordering.
+    """
+    org_age_when_awarded_buckets = context["results"]["aggregations"]["orgAgeWhenAwarded"]["buckets"]
+    ordered = []
+
+    def find_in_buckets(key):
+        for bucket in org_age_when_awarded_buckets:
+            if bucket["key"] == key:
+                return bucket
+
+    for age in AGE_BIN_LABELS:
+        age_found = find_in_buckets(age)
+        if age_found:
+            ordered.append(age_found)
+
+    context["results"]["aggregations"]["orgAgeWhenAwarded"]["buckets"] = ordered
+
+
+def redirect_request_to_include_all_org_ids(request):
+    """ If someone provides an org-id append all other known org-ids to the query
+    This is done so that it doesn't matter which org-id of many for a certain org
+    is provided we can still return the results.
+    """
+    do_redirect = False
+    request_get_copy = request.GET.copy()
+
+    for entity_type in [("fundingOrganization", "funder"), ("recipientOrganization", "recipient")]:
+        # Match the supplied org-id to any other org-ids in use
+        if org_ids := request.GET.getlist(entity_type[0]):
+            for org_id in org_ids:
+                try:
+                    for additional_org_id in get_org(org_id, entity_type[1])["orgIDs"]:
```

Comment:
Does this include the canonical org id (as well as "additional" org ids), in case they've searched for a non-canonical org? (if that's even possible here?)

Reply:
yeah the "orgIDs" field contains all the org-ids known for that org